### PR TITLE
Add New Tool xtuis: Message to Weixin/wechat

### PR DESCRIPTION
**Pull request by jimmy8952**
New Tool Name: xtuis
Description: send real time message to weixin/wechat if xtuis_api_key authorized.

# Description
Only Add one new tool name xtuis which can send real time message to weixin/wechat if individual xtuis_api_key is authorized.
while use LLM for chat, user sometimes want to send the LLM's answer directly to notify himself in weixin/wechat. LLM could use this xtuis tool to accomplish this task very easily. 

let me introduce xtuis a little bit, xtuis is completely free to use for anyone, it has been startup for more than 5 years and have thousands of user on field of Quantitative Investment, gaming,scraping for real time. everyone is free to have 300 times per day notification. by the way, i am the owner of xtuis.

there is no dependency for this changes, only one new tool xtuis is added.

Fixes # (issue)
None

## Type of Change
- [ ] New feature: Add new build in tool xtuis

# How Has This Been Tested?
it has been tested fully and test result is success.
Test Step:
1. Authorization of API Key: go to tools, click on xtuis icon, you need to fill in your xtuis_api_key.  to get the key, simple click on url, once you open website. the page give instruction how to get xtuis token, in short, you just open weixin to scan the QR code, immediately you will receive the xtuis token. here xtuis token is same as xtuis_api_key.  once you fill xtuis_api_key, save it.
2. Agent create: in stadio, create a agent with prompt as below:
   "you are brain of management and good at schedule a task, you prefer to use tools first get the information and analyse the result then answer user question"
   add tools: wikimedia, current time, xtuis_push(虾推啥).
3. choose llm chatgpt-3.5-turbo
4. start a new chat with below message:
"what is age of Biden who is current US president. pls use wikipedia tool to find date of birth, then use tool current time to get current time, then you can calculate the age. once you get final result, pls send original question and final result to weixin by using tool of xtuis_push"
5. you will see chat-gpt will use tool wikipedia to get date of birth, current time, then send weixin message to you. pls open you weixin, you will see a notification of age of Biden send to you. you can double click on this message card, then you will see full detail of the message.

- [ ] TODO

# Suggested Checklist:

- [ ] I have performed a self-review of my own code--Done.

### CODE REVIEW

**Review by Yeuoly — CHANGES_REQUESTED** [review]
hello, I think we should use a SVG icon instead, you can find the one you like here https://www.svgrepo.com @bowenliang123 has just recommended.

**Comment by bowenliang123** [comment]
@Yeuoly I'm afraid only popular IT giants' icon available on svgrepo.com. xtuis is one of them. It's okay to use PNG icon here.

File: api/core/tools/provider/builtin/xtuis/tools/xtuis_push.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+identity:
+  name: xtuis_push
+  author: Jimmy Wu
+  label:
+    en_US: Send Message to Weixin/Wechat
+    zh_Hans: 虾推啥-微信推送
```
Comment (bowenliang123) [inline api/core/tools/provider/builtin/xtuis/tools/xtuis_push.yaml]: 虾推啥 supports not only Wechat. Just use the name for this. And wechat support to be described in single function.
```suggestion
    zh_Hans: 虾推啥
```

**Comment by jimmy8952** [comment]
> hello, I think we should use a SVG icon instead, you can find the one you like here https://www.svgrepo.com @bowenliang123 has just recommended.

thanks, i have change to SVG icon format now. new commit has been submitted. pls kindly review again.

File: api/core/tools/provider/builtin/xtuis/xtuis.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  name: xtuis
+  label:
+    en_US: xtuis
+    zh_Hans: 虾推啥
+    pt_BR: xtuis
+  description:
+    en_US: send message to Weixin/Wechat
+    zh_Hans: 一行代码发送信息到微信
+    pt_BR: send message to Weixin/Wechat
+  icon: icon6.png
```
Comment (crazywoola) [inline api/core/tools/provider/builtin/xtuis/xtuis.yaml]: This should be `.svg`. 
Please rename `icon6.svg` to `xtuis.svg`.

**Review by crazywoola — CHANGES_REQUESTED** [review]

File: api/core/tools/provider/builtin/xtuis/xtuis.yaml (hunk 2 above)
Comment (jimmy8952) [reply api/core/tools/provider/builtin/xtuis/xtuis.yaml]: > This should be `.svg`. Please rename `icon6.svg` to `xtuis.svg`.

thanks. i have rename to xtuis.svg, and also update the xtuis.yaml with new icon names. pls kindly review it again.

**Comment by jimmy8952** [comment]
Hi Reviewer:

i prefer to upload the screenshot of my own testing to you, hope it more clearly for your testing.

Test Step:

1. Authorization of API Key: go to tools, click on xtuis icon, you need to fill in your xtuis_api_key. to get the key, simple click on url, once you open website. the page give instruction how to get xtuis token, in short, you just open weixin to scan the QR code, immediately you will receive the xtuis token. here xtuis token is same as xtuis_api_key. once you fill xtuis_api_key, save it.
dify1
dify2

2. Agent create: in stadio, create a agent with prompt as below:
"you are brain of management and good at schedule a task, you prefer to use tools first get the information and analyse the result then answer user question"
add tools: wikimedia, current time, xtuis_push(虾推啥).
choose llm chatgpt-3.5-turbo, then click publish.
dify3


3. start a new chat with below message:
"what is age of Biden who is current US president. pls use wikipedia tool to find date of birth, then use tool current time to get current time, then you can calculate the age. once you get final result, pls send original question and final result to weixin by using tool of xtuis_push"
dify4
dify5

you will see chat-gpt will use tool wikipedia to get date of birth, current time, then send weixin message to you. pls open you weixin, you will see a warning notification of age of Biden send to you. you can double click on this message card, then you will see full detail of the message.
dify6
dify7

**Comment by Yeuoly** [comment]
Sorry for that, but after a review of our operation department, this PR seems to have a tendency to direct traffic towards WeChat official accounts in a manner that might be more approachable and understandable, you can wait for your Tool store to be released, after that, you can upload any tool you want here.

**Comment by jimmy8952** [comment]
Ok, understand.

**Pull request closed without merging**